### ISSUE TITLE
XBUS - no data

### ISSUE THREAD

**Issue by jasc76** [pierrotm777/oXs-RP2040-i2c#1]
Hi

I tested with Spektrum DX9 and AR9030T and I get no telemetry values. FV shows baro but, I get no data. The reciever and stuff works, because with spektrum sensors, all works

**Comment by pierrotm777** [comment]
Thanks for your test.
I just received my own Spektrum receiver an AR6610T.
Do you have a picture of your connection ?

**Comment by pierrotm777** [comment]
When you type FVP or FVN in the serial console, do you see some change on your dx9 ?

**Comment by jasc76** [comment]
Sorry, didnt check the console. Just checked I have baro and its sending data

Picture, also no. But straight forward. like here: https://www.spektrumrc.com/ProdInfo/Files/SPM_Telemetry_Developers_Specs.pdf page 6
I know from MSRC, they use a transistor because they say the RP is not initializing fast enough.

**Comment by pierrotm777** [comment]
You say 'Just checked I have baro and its sending data' .
You receive data from the RP ?

**Comment by pierrotm777** [comment]
Can you try this uf2 version and see if the altitude is returned ?
[oXs.zip](https://github.com/pierrotm777/oXs-RP2040-i2c/files/12643341/oXs.zip)

**Comment by jasc76** [comment]
> You say 'Just checked I have baro and its sending data' .
> You receive data from the RP ?

No, I just checked with FV to see the RP sees data from baro

**Comment by jasc76** [comment]
> Can you try this uf2 version and see if the altitude is returned ?
> [oXs.zip](https://github.com/pierrotm777/oXs-RP2040-i2c/files/12643341/oXs.zip)

I will
but will take until next weekend 

I still fear its the same issue MSRC faced

**Comment by pierrotm777** [comment]
Did you have tried MSRC ?

**Comment by pierrotm777** [comment]
[oXs2.zip](https://github.com/pierrotm777/oXs-RP2040-i2c/files/12643480/oXs2.zip)
Another version, sorry.

**Comment by jasc76** [comment]
> Did you have tried MSRC ?

Yes, didnt work also. I did not mess with putting this transistor in. I put it aside and when you brought out this version I gave it another shot. Its not for me, a friend still flies Spektrum and I wanted to build it for him.

**Comment by jasc76** [comment]
> [oXs2.zip](https://github.com/pierrotm777/oXs-RP2040-i2c/files/12643480/oXs2.zip) Another version, sorry.

I just did a dry test. This version is not working properly. He says GPS not detected but it worked with your other version. FV just doesnt output anything

>> 
processing cmd

Cmd to execute:   FV

processing cmd

Cmd to execute:   FV

**Comment by pierrotm777** [comment]
GPS working with other version?
Latitude and longitude was good values? 
Ok, it's a good news.

**Comment by pierrotm777** [comment]
FV return true telemetry values.
Type FVP and FVN.

**Comment by jasc76** [comment]
> GPS working with other version? Latitude and longitude was good values? Ok, it's a good news.

GPS was working in oXs but not over XBUS ! Also Baro not working over XBUS

**Comment by jasc76** [comment]
> FV return true telemetry values. Type FVP and FVN.

Yes, and in your last test version https://github.com/pierrotm777/oXs-RP2040-i2c/issues/1#issuecomment-1722540158 FV doesnt return values....  In the first version FV worked

It was just an obersavation..... I thought FV should always work !?

**Comment by pierrotm777** [comment]
FV work always but return the good telemetry from oXs. FVP and FVN return always positives or negatives fixed values.

**Comment by jasc76** [comment]
yes, I know
again, your latest test version from this thread  is not working at all  ! Fv is empty

**Comment by Satcomix** [comment]
Hello Pierre,
I encountered a problem with the SRXL2 protocol at RCG and wanted to do some experiments with my old Spektrum TM1000 and DX8 via I2C/XBUS.
But the XBUS-oXs-Version doesn't work.
SDA0=TLM=GPio4,
SCL0=PRI=GPio5
Do you have any idea what it could be?
Best regards,
Torsten

**Comment by pierrotm777** [comment]
Hello Satcomix,
The XBUS version for oXs isn't ready and don't run for now.
I am not a good programmer and don't find how to adapt the examples i have found.

**Comment by Satcomix** [comment]
Hello Pierre,
I am not a good programmer too. :-))
I took a look at the srxl2.h/.cpp and srxl_sensors.h.
I only found 5 sensors that transmit the telemetry data.
Vario_Simple, GPS, Airspeed, ESC, RX_MAH
I will ask Mstrens when we can do tests for the SRXL2 protocol again, as he is currently busy with the gyro system.
If the SRXL2 runs properly, we will then build the XBUS together.
greetings,
Torsten

**Comment by pierrotm777** [comment]
SRXL2 is ran very well during my old tests.
Xbus need an i2c input protocol.
I have done some tests but I receive some bad info on my transmitter.

**Comment by Satcomix** [comment]
> SRXL2 is ran very well during my old tests. Xbus need an i2c input protocol. I have done some tests but I receive some bad info on my transmitter.

Hello Pierre, can you make one more test with the SRXL2 protocol and the latest version 2.10.8-test (and post it), as I cannot receive all sensor IDs on my old DX8.
If you can receive all telemetry data, then the problem must be with AKob (Arnd).
Greetings,
Torsten

**Comment by pierrotm777** [comment]
Ok, I shall do a test tomorrow 😉

**Comment by Satcomix** [comment]
> Ok, I shall do a test tomorrow 😉

Thank you very much for the test with SRXL2.
I will ask Michel for the possibilty of XBUS telemetry for oXs_RP2040. :-))
greetings,
Torsten

**Comment by pierrotm777** [comment]
Ok, this is the results of my SRXL2 tests.
For theses tests, i use a TX16 MkII MultiProtocol DSMX Auto with the 2.10.8 version with the original uf2 firmware.
I use EdgeTx 2.8.5, may be i should use 2.9.2.
I have tried 2 receivers.
- SPM4648 (sattelite receiver).
- AR6610T.
I had done my old first tests with a DX8 and the SPM4648 with success, if i remember, i used the 2.8.5 .
But now, i receive only:
TRSS 31
FdeA 7
FLss 0
Hold 0

With the AR6610T i receive more values:
TRSS 31
FdeA 7
FLss 0
Hold 0
A2 5.06v
Alt 0.4m
Vspd 0m/s

No GPS

**Comment by pierrotm777** [comment]
Idem with 2.9.2.

**Comment by Satcomix** [comment]
> Ok, this is the results of my SRXL2 tests. For theses tests, i use a TX16 MkII MultiProtocol DSMX Auto with the 2.10.8 version with the original uf2 firmware. I use EdgeTx 2.8.5, may be i should use 2.9.2. I have tried 2 receivers.
> 
>     * SPM4648 (sattelite receiver).
> 
>     * AR6610T.
>       I had done my old first tests with a DX8 and the SPM4648 with success, if i remember, i used the 2.8.5 .
>       But now, i receive only:
>       TRSS 31
>       FdeA 7
>       FLss 0
>       Hold 0
> 
> 
> With the AR6610T i receive more values: TRSS 31 FdeA 7 FLss 0 Hold 0 A2 5.06v Alt 0.4m Vspd 0m/s
> 
> No GPS

Hello Pierre,
Thank you for your test. Your result also agrees with mine and Arnd's.
So I will reopen the older SRXL2 issue.
**Edit: I have reopened the issue.**
You are cordially invited to test it, four eyes see more than two.
Thank you again and greetings,
Torsten

**Comment by Satcomix** [comment]
Hello Pierre,
Mstrens has released a new version 2.10.10-test, where there were changes regarding SRXL2.
I now receive more data with my self-built version for the old DX8
Could you test this version again? 
If everything works now, I can close the issue again.
greetings,
Torsten

**Comment by pierrotm777** [comment]
I have already tried it with the same result.
I have a new issue with all my gps ublox witch don't run.
U-Blox center see this m8n. Curious.

**Comment by Satcomix** [comment]
> I have already tried it with the same result. I have a new issue with all my gps ublox witch don't run. U-Blox center see this m8n. Curious.

You dont see the telemetry values,except the four values you hve seen before.???
I have Beitian BE-250 M10 Chipset at my oXs_RP2040 in GPS=U Mode. No Problems.

**Comment by pierrotm777** [comment]
Only these values and no gps with ALL protocols !
TRSS 31
FdeA 7
FLss 0
Hold 0
A2 5.06v
Alt 0.4m
Vspd 0m/s

**Comment by pierrotm777** [comment]
I try to configure only my gps gps_rx=13 et gps_tx=12 but its not responding 
```
Version = 2.10.10 
    Function                GPIO  Change entering XXX=yyy (yyy=255 to disable)
Primary channels input    =  255  (PRI     = 5, 9, 21, 25)
Secondary channels input  =  255  (SEC     = 1, 13, 17, 29)
Telemetry . . . . . . . . =    8  (TLM     = 0, 1, 2, ..., 29)
GPS Rx  . . . . . . . . . =   13  (GPS_RX  = 0, 1, 2, ..., 29)
GPS Tx  . . . . . . . . . =   12  (GPS_TX  = 0, 1, 2, ..., 29)
Sbus OUT  . . . . . . . . =  255  (SBUS_OUT= 0, 1, 2, ..., 29)
RPM   . . . . . . . . . . =  255  (RPM     = 0, 1, 2, ..., 29)
SDA (I2C sensors) . . . . =  255  (SDA     = 2, 6, 10, 14, 18, 22, 26)
SCL (I2C sensors) . . . . =  255  (SCL     = 3, 7, 11, 15, 19, 23, 27)
PWM Channels 1, 2, 3 ,4   =  255  255  255  255 (C1 / C16= 0, 1, 2, ..., 15)
PWM Channels 5, 6, 7 ,8   =  255  255  255  255
PWM Channels 9,10,11,12   =  255  255  255  255
PWM Channels 13,14,15,16  =  255  255  255  255
Voltage 1, 2, 3, 4        =  255  255  255  255 (V1 / V4 = 26, 27, 28, 29)
RGB led . . . . . . . . . =   16  (RGB    = 0, 1, 2, ..., 29)
Logger  . . . . . . . . . =  255  (LOG    = 0, 1, 2, ..., 29)
ESC . . . . . . . . . . . =  255  (ESC_PIN= 0, 1, 2, ..., 29)Esc type is not defined

Protocol is Sport (Frsky)
CRSF baudrate   = 420000
Logger baudrate = 115200
PWM is generated at = 50 Hz
Voltage parameters:
    Scales : 1.000000 , 1.000000 , 1.000000 , 1.000000 
    Offsets: 0.000000 , 0.000000 , 0.000000 , 0.000000 
    No temperature sensors are connected on V3 and V4
RPM multiplier = 1.000000
Baro sensor is not detected
Airspeed sensor is not detected
    No Vspeed compensation channel defined; oXs uses default settings
First analog to digital sensor is not detected
Second analog to digital sensor is not detected
Foreseen GPS type is Ublox (configured by oXs) :GPS is not (yet) detected
Led color is normal (not inverted)
Failsafe type is HOLD
Acc/Gyro is not detected

Gyro is not configured

No sequencers are defined

Config parameters are OK
Press ? + Enter to get help about the commands
```

**Comment by Satcomix** [comment]
Protocol S.PORT??? for the GPS Test. Ahh, now i understand
I will make the same test with my radio and S.PORT and GPS

**Comment by Satcomix** [comment]
**Everything works with S.PORT !!!**

processing cmd

Version = 2.10.10 
    Function                GPIO  Change entering XXX=yyy (yyy=255 to disable)
Primary channels input    =    5  (PRI     = 5, 9, 21, 25)
Secondary channels input  =  255  (SEC     = 1, 13, 17, 29)
Telemetry . . . . . . . . =    1  (TLM     = 0, 1, 2, ..., 29)
GPS Rx  . . . . . . . . . =   12  (GPS_RX  = 0, 1, 2, ..., 29)
GPS Tx  . . . . . . . . . =   13  (GPS_TX  = 0, 1, 2, ..., 29)
Sbus OUT  . . . . . . . . =    2  (SBUS_OUT= 0, 1, 2, ..., 29)
RPM   . . . . . . . . . . =    4  (RPM     = 0, 1, 2, ..., 29)
SDA (I2C sensors) . . . . =   14  (SDA     = 2, 6, 10, 14, 18, 22, 26)
SCL (I2C sensors) . . . . =   15  (SCL     = 3, 7, 11, 15, 19, 23, 27)
PWM Channels 1, 2, 3 ,4   =  255  255  255  255 (C1 / C16= 0, 1, 2, ..., 15)
PWM Channels 5, 6, 7 ,8   =    8    9  255  255
PWM Channels 9,10,11,12   =  255  255  255  255
PWM Channels 13,14,15,16  =  255  255   10   11
Voltage 1, 2, 3, 4        =   26   27   28   29 (V1 / V4 = 26, 27, 28, 29)
RGB led . . . . . . . . . =   18  (RGB    = 0, 1, 2, ..., 29)
Logger  . . . . . . . . . =    0  (LOG    = 0, 1, 2, ..., 29)
ESC . . . . . . . . . . . =  255  (ESC_PIN= 0, 1, 2, ..., 29)    Esc type is HW4 (Hobbywing V4)

Protocol is Sport (Frsky)
CRSF baudrate   = 420000
Logger baudrate = 115200
PWM is generated at = 50 Hz
Voltage parameters:
    Scales : 1.000000 , 1.000000 , 1.000000 , 1.000000 
    Offsets: 0.000000 , 0.000000 , 2.000000 , 0.000000 
    Temperature sensors are connected on V3 and V4
RPM multiplier = 1.000000
Baro sensor is detected using MS5611
    Sensitivity min = 100 (at 100)   , max = 300 (at 1000)
    Hysteresis = 5 
Aispeed sensor is detected using MS4525
    Vspeed compensation channel = 7
First analog to digital sensor is detected using ads1115
    Measurement setup: 4 , 5 , 6 ,7
    Gains: 1 , 1 , 1 ,1
    Rates: 5 , 5 , 5 ,5
    Offsets: 0.000000 , 0.000000 , 0.000000 ,0.000000
    Scales: 1.000000 , 1.000000 , 1.000000 ,1.000000
    Averaged on: 10 , 10 , 10 ,10
Second analog to digital sensor is detected using ads1115
    Measurement setup: 4 , 5 , 6 ,7
    Gains: 1 , 1 , 1 ,1
    Rates: 5 , 5 , 5 ,5
    Offsets: 0.000000 , 0.000000 , 0.000000 ,0.000000
    Scales: 1.000000 , 1.000000 , 1.000000 ,1.000000
    Averaged on: 10 , 10 , 10 ,10
Foreseen GPS type is Ublox (configured by oXs) :GPS is detected and has a fix
Led color is normal (not inverted)
Failsafe uses predefined values
     Chan 1...4  =  1500  1500  1500  1500
     Chan 5...8  =  2012  2011  1500  1500
     Chan 9...12 =  1500  1500  1500  1500
     Chan 13...16=  1500  1500  1500  1500
Acc/Gyro is detected using MP6050
     Acceleration offsets X, Y, Z = 942 , 159 , -2670
     Gyro offsets         X, Y, Z = 2 , -8 , -12
     Orientation          Horizontal is UP(Z+)     Vertical is FRONT(X+)

Gyro is not configured

No sequencers are defined

Config parameters are OK
Press ? + Enter to get help about the commands
processing cmd

Cmd to execute:   FV

GPS Latitude = XX.X225440 degree
GPS Longitude = X.X173744 degree
GPS Groundspeed = 13 cm/s
GPS Heading = 75.610000 degree
GPS Altitude = 4981 cm
GPS Num sat. = 118
GPS Date J M A = 21 11 23 
GPS Time H M S = 15 29 40 
GPS Pdop = 154 
GPS Home bearing = 345 degree
GPS Home distance = 3 m
Volt 1 = 1555 mVolt
Current (Volt 2) = 1812 mA
Capacity (using current) = 17 mAh
Temp 1 (Volt 3) = 228 degree
Temp 2 (Volt 4) = 1601 degree
Vspeed = -1 cm/s
Baro Rel altitude = -2 cm
Pitch = 0.000000 degree
Roll = 0.000000 degree
RPM = 999 Hertz
Ads 1 1 = 119 mVolt
Ads 1 2 = 119 mVolt
Ads 1 3 = 119 mVolt
Ads 1 4 = 119 mVolt
Ads 2 1 = 118 mVolt
Ads 2 2 = 119 mVolt
Ads 2 3 = 118 mVolt
Ads 2 4 = 119 mVolt
Airspeed = 3 cm/s
Compensated Vspeed = -12 cm/s
Gps cumulative distance = 6
Acc X = -0.009000g
Acc Y = -0.002000g
Acc Z = 0.993000g
Vspeed compensation = 1.10
pwmTop= 20000

**Comment by pierrotm777** [comment]
Ok, i don't know why but now my gps is ok with Frsky after a new bind.

**Comment by pierrotm777** [comment]
New version I2C include now Radiolink, Hitec and Spektrum Xbus
